bodhi/tests/server/scripts/test_sar.py

```diff
@@ -0,0 +1,150 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+This module contains tests for the bodhi.server.scripts.sar module.
+"""
+import os
+from datetime import datetime
+
+from click import testing
+import mock
+
+from bodhi.server import models
+from bodhi.server.scripts import sar
+from bodhi.tests.server.base import BaseTestCase
+
+
+EXPECTED_USER_DATA_OUTPUT = """\
+==========> User account data for: guest <==========
+
+email: None
+groups: ['packager']
+popups: True
+
+----> Comments: <----
+
+Comment no 1:
+anonymous: False
+karma: 1
+karma_critpath: 0
+text: wow. amaze.
+timestamp: 1984-11-02 00:00:00
+update_alias: FEDORA-2018-a3bbe1a8f2
+username: guest
+
+----> Updates: <----
+
+Update no 1:
+alias: FEDORA-2018-a3bbe1a8f2
+autokarma: True
+bugs: [12345]
+builds: ['bodhi-2.0-1.fc17']
+close_bugs: True
+date_submitted: 1984-11-02 00:00:00
+notes: Useful details!
+release_name: F17
+require_bugs: False
+require_testcases: False
+requirements: rpmlint
+severity: medium
+stable_karma: 3
+suggest: unspecified
+type: bugfix
+unstable_karma: -3
+user: guest
+"""
+
+EXPECTED_JSON_OUTPUT = (
+    '{"guest": {"comments": [{"anonymous": false, "karma": 1, "karma_critpath": 0, "text":'
+    ' "wow. amaze.", "timestamp": "1984-11-02 00:00:00", "update_alias": '
+    '"FEDORA-2018-a3bbe1a8f2", "username": "guest"}], "email": null, "groups": '
+    '["packager"], "name": "guest", "show_popups": true, "updates": [{"alias": '
+    '"FEDORA-2018-a3bbe1a8f2", "autokarma": true, "bugs": [12345], "builds": '
+    '["bodhi-2.0-1.fc17"], "close_bugs": true, "date_submitted": "1984-11-02 00:00:00", '
+    '"notes": "Useful details!", "release_name": "F17", "require_bugs": false, '
+    '"require_testcases": false, "requirements": "rpmlint", "severity": "medium", '
+    '"stable_karma": 3, "suggest": "unspecified", "type": "bugfix", '
+    '"unstable_karma": -3, "user": "guest"}]}}\n'
+)
+
+
+class TestSar(BaseTestCase):
+    """This class contains tests for the get_user_data() function."""
+
+    def test_invalid_user(self):
+        """Ensure nothing is pirinted when user is not found and human readable is off."""
+        runner = testing.CliRunner()
+        r = runner.invoke(sar.get_user_data, ["--username=" + "invalid_user"])
+
+        self.assertEqual(r.exit_code, 0)
+        self.assertEqual(r.output, "")
```

Comment:
I think this is not standard compliant: it should, at least, return an empty set `{}`.
In my opinion the best option would be a structure like
```
{
  "error": null,
  "result": {<USER_DATA>}
}
```
for successful retrieval, and
```
{
  "error": {
    "message": "User not found."
  },
  "result": null
}
```
for the "user not found" error. In this case the output could also handle database connection error messages.

Reply:
For the record, the script that this is intended to replace also prints nothing, so I think this is fine.